### ISSUE TITLE
DanmNets delegated to other CNI plugins are always named "eth0"

### ISSUE THREAD

**Issue by uablrek** [nokia/danm#25]
I have 2 DamnNets delegating to other CNI plugins;
```
apiVersion: danm.k8s.io/v1
kind: DanmNet
metadata:
  name: default
spec:
  NetworkID: default
  NetworkType: bridge
---
apiVersion: danm.k8s.io/v1
kind: DanmNet
metadata:
  name: ipvlan-orig
spec:
  NetworkID: ipvlan-orig
  NetworkType: ipvlan-orig
  container_prefix: ipv
```

Where "bridge" is the main k8s network that should be "eth0" and the "ipvlan-orig" is a renamed CNI "standard" plugin (since I can't use the built-in with cri-o #24 ).

I start a alpine pod with;
```
      annotations:
        danm.k8s.io/interfaces: |
          [
            {
              "network":"default",
              "network":"ipvlan-orig"
            }
          ]
```
and enter an alpine pod with;
```
kubectl exec -it alpine-deployment-d4c99889-6znws sh
```

I expect to see the `eth0` interface connected to the bridge network and an extra interface named `ipv0` (or something similar) connected to the ipvlan but instead there is only a `eth0` interface and that is the ipvlan network. So it seems all delegated CNI plugins compete for the `eth0` name and the latest wins.

**Comment by uablrek** [comment]
I think you must make the delegation work. The "popular CNIs" today may not be popular tomorrow and new ones will become popular. You will never keep up :-)

Anyway the fix for #24 does not seem to hard so maybe I wait for that and test the built-in "ipvlan".

**Comment by Levovar** [comment]
Actually delegation does work, and work as intended.
DANM invokes the delegate plugins transparently, and does not interfere with their internals. Same idea of how metaplugins should act.

Now, regardless of how metas "should" act, we do like to interfere so an enhancement is already planned addressing this: https://github.com/nokia/danm/issues/15
In the scope of this enhancement we will internally keep track of interface names, and instruct delegated CNIs to set-up their network interfaces with a unique name.

I'm closing this issue with a duplicate if you don't mind!
